gzhttp/compress.go

```diff
@@ -464,6 +464,12 @@ func NewWrapper(opts ...option) (func(http.Handler) http.HandlerFunc, error) {
 	return func(h http.Handler) http.HandlerFunc {
 		return func(w http.ResponseWriter, r *http.Request) {
 			w.Header().Add(vary, acceptEncoding)
+			if c.allowCompressedRequests && contentGzip(r) {
+				r.Header.Del(contentEncoding)
+				r.Body = io.NopCloser(&gzipReader{body: r.Body})
+				defer r.Body.Close()
```

Comment:
```suggestion
				defer r.Close()
```
With the change above and this the body is always closed and decoder is recycled.

Reply:
@klauspost, ``defer r.Close`` - not found this function, maybe we should leave the old version ``defer r.Body.Close()``?

Reply:
Ah. Yeah. Correct.